lib/service/moduleGenerator.js

```diff
@@ -101,15 +98,12 @@ function generateModules(task) {
     containerId: task.containerId
   };
 
-  requestMetadata.securityContext = getSecurityContextString(
-    getProp(task.request.headers, ['authorization'], []),
-    appMetadata
-  );
+  requestMetadata.securityContext = getSecurityContextString(task.request.headers.authorization, appMetadata);
 
   let useBSONObjectId = false;
-  const migrated = getProp(task, ['appMetadata', 'appMetadata', 'objectid_migration', 'status']);
-  if (migrated != null && migrated === 'done') {
-    useBSONObjectId = true;
+
+  if (task.appMetadata.maintenance && task.appMetadata.maintenance.objectid_migration) {
```

Comment:
The path on this line has changed. Was the original path a bug? `getProp(task, ['appMetadata', 'appMetadata', 'objectid_migration', 'status'])`

Reply:
Yes, it was a bug.